packages/astro/src/integrations/hooks.ts

```diff
@@ -126,9 +126,9 @@ export async function runHookConfigSetup({
 	isRestart?: boolean;
 	fs?: typeof fsMod;
 }): Promise<AstroSettings> {
-	// An adapter is an integration, so if one is provided push it.
+	// An adapter is an integration, so if one is provided add it to the list of integrations.
 	if (settings.config.adapter) {
-		settings.config.integrations.push(settings.config.adapter);
+		settings.config.integrations.unshift(settings.config.adapter);
```

Comment:
This is a potentially consequential change, but it's necessary now because an adapter can force change the build output. Before, if an adapter changed the output to be a server, if an integration tried to read the buildOutput before the adapter was added, the value would be incorrect.

Reply:
I don't think `.push` was done for any intentional reason, but it is possible that someone is relying on this. We'll see.

Reply:
Hi @Princesseuh, I think this change affects the `@astro/sitemap` integration. I found #12663 where the sitemap is generated after the `@astro/vercel` adapter copies the `dist` files over to the `.vercel/output` directory, so the sitemap files don't end up being copied over to `.vercel/output/static`.

I did a quick test and manually modified this line in my `node_modules` to change it back to `push` and it copies the sitemap files over correctly.